client/core/core.go

```diff
@@ -1395,7 +1400,22 @@ func (c *Core) Network() dex.Network {
 // Exchanges creates a map of *Exchange keyed by host, including markets and
 // orders.
 func (c *Core) Exchanges() map[string]*Exchange {
-	return c.exchangeMap()
+	dcs := c.dexConnections()
+	infos := make(map[string]*Exchange, len(dcs))
+	for _, dc := range dcs {
+		infos[dc.acct.host] = dc.exchangeInfo()
+	}
+	return infos
+}
+
+// Exchange returns an exchange with a certain host. It returns an error if
+// no exchange exists at that host.
+func (c *Core) Exchange(host string) (*Exchange, error) {
+	exchange, ok := c.Exchanges()[host]
```

Comment:
The point of the new method is to avoid all the extra calls to `exchangeInfo` for hosts we don't care about.


```Go
func (c *Core) Exchange(host string) (*Exchange, error) {
	dc, _, err := c.dex(host)
	if err != nil {
		return nil, err
	}
	return dc.exchangeInfo(), nil
}
```

Reply:
Haha my bad.